### DISCUSSION TITLE
When clearing non-primes in the sieve (the second operation), the algorithm clears all non-primes individually, increasing the number with 2 * factor on each cycle.

### DISCUSSION THREAD

**Discussion by schmide** [PlummersSoftwareLLC/Primes#454] [Q&A] [answered]
Can someone clarify to what extent "individually" is enforced? 

Does this mean grouping or caching writes is not permitted?

Technically removing the even numbers is a violation of this rule.

**Comment by rbergen** [comment, marked as answer]
What matters for sure is that all the calculations and indexing related to clearing the prime candidate flags (whether they are booleans, bits or another data type) are performed during the "second operation" within the timed algorithm, and not based on pre-calculated bitmaps or something similar. Beyond that, as with any rule, we make up our minds about an approach when there is actual source code to look at.

**Comment by schmide** [comment]
Thanks. 

Just to clarify I was going to write bits in the second pass into quadwords before storing them one at a time.
